core-bundle/tests/Search/Backend/BackendSearchTest.php

```diff
@@ -375,3 +402,7 @@ private function createWebworkerWithCliRunning(): WebWorker
         return $webWorker;
     }
 }
+
+abstract class DocumentProvider implements ProviderInterface, TagProvidingProviderInterface
+{
+}
```

Comment:
This should be a fixture or an anonymous class.

Reply:
Extracted into fixture. I just copied this from how Symfony does it. Feels cleaner if it's just used in this very test class instead of having random classes floating around.
An anonymous class is not an option because I need to mock it.